### PR TITLE
Bump kotlin and kotlin gradle plugins

### PR DESCRIPTION
**Pull request by SultanArshad**

### CODE REVIEW

File: app-kotlin/build.gradle [hunk 1]
```diff
@@ -5,37 +5,46 @@ buildscript {
...
 apply plugin: 'com.android.application'
 apply plugin: 'kotlin-android'
 
 android {
-  compileSdkVersion 28
-  buildToolsVersion '28.0.3'
+  compileSdkVersion 31
 
-  defaultConfig {
+    defaultConfig {
```
Comment (xxfast) [inline app-kotlin/build.gradle:18]: Looks like you've introduced a space here, is this intended?

File: app-kotlin/build.gradle [hunk 2]
```diff
@@ -5,37 +5,46 @@ buildscript {
...
     }
   }
 
-  sourceSets {
-    main.java.srcDirs += 'src/main/kotlin'
+  compileOptions {
+    sourceCompatibility JavaVersion.VERSION_1_8
+    targetCompatibility JavaVersion.VERSION_1_8
+  }
+  kotlinOptions {
```
Comment (xxfast) [inline app-kotlin/build.gradle:37]: According to [the style guide](https://github.com/Gridstone/KotlinStyleGuide), we need a line break between blocks

File: gradle.properties [hunk 3]
```diff
@@ -15,4 +15,5 @@
 # When configured, Gradle will run in incubating parallel mode.
 # This option should only be used with decoupled projects. More details, visit
 # http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
-# org.gradle.parallel=true
\ No newline at end of file
+# org.gradle.parallel=true
+android.useAndroidX=true
```
Comment (xxfast) [inline gradle.properties:19]: Make sure to enable this setting to add a line break at the end of each file
Screen Shot 2022-06-30 at 10 50 08 am

File: app-kotlin/build.gradle [hunk 4]
```diff
@@ -5,37 +5,46 @@ buildscript {
...
 dependencies {
-  implementation 'org.jetbrains.kotlin:kotlin-stdlib:1.3.21'
-  implementation 'androidx.appcompat:appcompat:1.0.2'
+  implementation 'androidx.core:core-ktx:1.7.0'
+  implementation 'androidx.appcompat:appcompat:1.3.0'
+  implementation 'com.google.android.material:material:1.4.0'
+  implementation 'androidx.constraintlayout:constraintlayout:2.0.4'
+  testImplementation 'junit:junit:4.13.2'
+  androidTestImplementation 'androidx.test.ext:junit:1.1.3'
+  androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
```
Comment (xxfast) [inline app-kotlin/build.gradle:49]: do we need all these dependencies?

**Review by xxfast — CHANGES_REQUESTED** [review]
Nice work 👍

File: app-kotlin/build.gradle (hunk 1 above)
Comment (SultanArshad) [reply app-kotlin/build.gradle:18]: thanks for mentioning i think this is not intended.

File: app-kotlin/build.gradle (hunk 2 above)
Comment (SultanArshad) [reply app-kotlin/build.gradle:37]: Thanks from now i will follow style guide

File: gradle.properties (hunk 3 above)
Comment (SultanArshad) [reply gradle.properties:19]: Thanks i did the changes in studio settings

File: app-kotlin/build.gradle (hunk 4 above)
Comment (SultanArshad) [reply app-kotlin/build.gradle:49]: Not sure but might needed in future i will remove for now and added when required

**Review by xxfast — APPROVED** [review]
Nice work 👍

**Pull request merged**